test/DualQuaternion.jl

```diff
@@ -83,38 +197,228 @@ using Test
     end
 
     @testset "basic" begin
-        # TODO: test real, imag, conj, dconj, float, and Quaternions.abs_imag
+        q = rand(DualQuaternionF64)
+        qnorm = normalize(q)
+        @test_throws MethodError imag(q)
+        @test conj(q) === dualquat(conj(q.q0), conj(q.qe), q.norm)
+        @test conj(qnorm) === dualquat(conj(qnorm.q0), conj(qnorm.qe), qnorm.norm)
+        @test conj(conj(q)) === q
+        @test conj(conj(qnorm)) === qnorm
+        @test float(dualquat(dual.(1:4, 5:8)...)) === dualquat(dual.(1.0:4.0, 5.0:8.0)...)
+        @test dconj(q) === dualquat(q.q0, -q.qe, q.norm)
+        @test dconj(qnorm) === dualquat(qnorm.q0, -qnorm.qe, qnorm.norm)
+        @test Quaternions.Q0(q) === q.q0
+        @test Quaternions.Qe(q) === q.qe
     end
 
-    @testset "algebraic properties" begin end
-
-    @testset "isreal" begin end
-
-    @testset "iszero" begin end
-
-    @testset "isone" begin end
-
-    @testset "isfinite" begin end
-
-    @testset "isinf" begin end
-
-    @testset "isnan" begin end
+    @testset "algebraic properties" begin
+        @testset "addition/subtraction" begin
+            for _ in 1:100
+                dq1, dq2 = rand(DualQuaternionF64, 2)
+                @test (dq1 + dq2).q0 ≈ dq1.q0 + dq2.q0
+                @test (dq1 + dq2).qe ≈ dq1.qe + dq2.qe
+                @test (dq1 - dq2).q0 ≈ dq1.q0 - dq2.q0
+                @test (dq1 - dq2).qe ≈ dq1.qe - dq2.qe
+                @test (+dq1).q0 === dq1.q0
+                @test (+dq1).qe === dq1.qe
+                @test (-dq1).q0 === -dq1.q0
+                @test (-dq1).qe === -dq1.qe
+            end
+        end
+        @testset "division" begin
+            for _ in 1:100
+                dq = rand(DualQuaternionF64)
+                dq2 = dq / dq
+                @test dq2.q0 ≈ 1
+                @test dq2.qe ≈ zero(dq2.qe) atol = 1e-6
+                dq2 = dq \ dq
+                @test dq2.q0 ≈ 1
+                @test dq2.qe ≈ zero(dq2.qe) atol = 1e-6
+                s = randn()
+                @test (dq / s).q0 ≈ dq.q0 / s
+                @test (dq / s).qe ≈ dq.qe / s
+                d = dual(randn(2)...)
+                @test (dq / d).q0 ≈ (dq / dualquat(d)).q0
+                @test (dq / d).qe ≈ (dq / dualquat(d)).qe
+            end
+        end
+        @testset "multiplication is associative" begin
+            for _ in 1:100
+                dq1, dq2, dq3 = rand(DualQuaternionF64, 3)
+                dq4 = (dq1 * dq2) * dq3
+                dq5 = dq1 * (dq2 * dq3)
+                @test dq4.q0 ≈ dq5.q0
+                @test dq4.qe ≈ dq5.qe
+            end
+        end
+        @testset "multiplication by dual" begin
+            for _ in 1:10
```

Comment:
```suggestion
            for _ in 1:100
```

Also for this line: https://github.com/sethaxen/Quaternions.jl/blob/moretests/test/Quaternion.jl#L191